lib/plugins/aws/utils/parseS3URI.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const patterns = [
+  // S3 URI. Ex: s3://bucket/path/to/artifact.zip
+  new RegExp('^s3://([^/]+)/(.+)'),
+
+  // New style S3 URL. Ex: https://bucket.s3.amazonaws.com/path/to/artifact.zip
+  new RegExp('([^/]+)\\.s3\\.amazonaws\\.com/(.+)'),
+
+  // Old style S3 URL. Ex: https://s3.amazonaws.com/bucket/path/to/artifact.zip
+  new RegExp('s3\\.amazonaws\\.com/([^/]+)/(.+)'),
+
+  // ARN. Ex: arn:aws:s3:::bucket/path/to/artifact.zip
```

Comment:
I'm not sure about support for `arn` here. The name of the util is also misleading, as `arn` is not a URI in this case

Reply:
I agree, it'll be good to clarify the use case. If it's just about _nice to have_ let's not add it.

Reply:
I was also on the fence about including it. I only added it because it was the only other S3 string format not being supported so I included it for completion purposes. I'll remove it.